services/registration/src/routes/application.ts

```diff
@@ -224,6 +225,26 @@ applicationRouter.route("/:id/actions/submit-application").post(
       throw new BadRequestError("Unable to submit application data. Branch is not currently open.");
     }
 
+    const applicantType = branch.applicationGroup;
+    const autoConfirm = branch.automaticConfirmation;
+
+    if (
+      autoConfirm?.enabled &&
+      (applicantType === "PARTICIPANT" || existingApplication.email in autoConfirm.emails!)
+    ) {
```

Comment:
Also, this functionality only checks if the email matches the list. It should also support matching an email domain, or matching all emails (this is when one of the fields is a "*". Additionally, you should really try to avoid using the bang operator (`autoConfirm.emails!`), instead use something like (`autoConfirm.emails ?? []`) to prevent errors that may occur